pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -688,6 +688,198 @@ func GetRecordingRules(namespace string) []KubevirtRecordingRule {
+				Expr:   intstr.FromString("kubevirt_vmi_memory_swap_in_traffic_bytes"),
+			},
+			MType:       prometheusv1.MetricTypeGauge,
+			Description: "The total amount of data read from swap space of the guest in bytes. [Deprecated]",
+		},
+		{
+			Rule: v1.Rule{
+				Record: "kubevirt_vmi_memory_swap_out_traffic_bytes_total",
+				Expr:   intstr.FromString("kubevirt_vmi_memory_swap_out_traffic_bytes"),
+			},
+			MType:       prometheusv1.MetricTypeGauge,
+			Description: "The total amount of memory written out to swap space of the guest in bytes. [Deprecated]",
+		},
+		{
+			Rule: v1.Rule{
+				Record: "kubevirt_vmi_outdated_count",
+				Expr:   intstr.FromString("kubevirt_vmi_number_of_outdated"),
+			},
+			MType:       prometheusv1.MetricTypeGauge,
+			Description: "Indication for the total number of VirtualMachineInstance workloads that are not running within the most up-to-date version of the virt-launcher environment. [Deprecated]",
+		},
+		{
+			Rule: v1.Rule{
+				Record: "kubevirt_vmi_storage_flush_times_ms_total",
+				Expr:   intstr.FromString("kubevirt_vmi_storage_flush_times_seconds_total * 1000"),
+			},
+			MType:       prometheusv1.MetricTypeCounter,
+			Description: "Total time (ms) spent on cache flushing. [Deprecated]",
+		},
+		{
+			Rule: v1.Rule{
+				Record: "kubevirt_vmi_storage_read_times_ms_total",
+				Expr:   intstr.FromString("kubevirt_vmi_storage_read_times_ms_total * 1000"),
+			},
+			MType:       prometheusv1.MetricTypeCounter,
+			Description: "Total time (ms) spent on read operations. [Deprecated]",
+		},
+		{
+			Rule: v1.Rule{
+				Record: "kubevirt_vmi_storage_write_times_ms_total",
+				Expr:   intstr.FromString("kubevirt_vmi_storage_write_times_seconds_total * 1000"),
+			},
+			MType:       prometheusv1.MetricTypeCounter,
+			Description: "Total time (ms) spent on write operations. [Deprecated]",
+		},
+		{
+			Rule: v1.Rule{
+				Record: "kubevirt_vmi_vcpu_seconds",
+				Expr:   intstr.FromString("kubevirt_vmi_vcpu_seconds_total"),
+			},
+			MType:       prometheusv1.MetricTypeCounter,
+			Description: "Total amount of time spent in each state by each vcpu (cpu_time excluding hypervisor time). Where `id` is the vcpu identifier and `state` can be one of the following: [`OFFLINE`, `RUNNING`, `BLOCKED`]. [Deprecated]",
+		},
+		{
+			Rule: v1.Rule{
+				Record: "kubevirt_vmi_vcpu_wait_seconds",
+				Expr:   intstr.FromString("kubevirt_vmi_vcpu_wait_seconds_total"),
+			},
+			MType:       prometheusv1.MetricTypeCounter,
+			Description: "Amount of time spent by each vcpu while waiting on I/O. [Deprecated]",
+		},
+		{
+			Rule: v1.Rule{
+				Record: "kubevirt_vmsnapshot_disks_restored_from_source_total",
+				Expr:   intstr.FromString("kubevirt_vmsnapshot_disks_restored_from_source"),
+			},
+			MType:       prometheusv1.MetricTypeGauge,
+			Description: "Returns the total number of virtual machine disks restored from the source virtual machine. [Deprecated]",
+		},
```

Comment:
maybe can we move these to a separate function, maybe `GetDeprecatedRecordingRules`, to have a clear distinction for the deprecated items?

maybe even somehow include the last version they were not deprecated -> `GetDeprecatedRecordingRulesAfterRelease59`

that would make it easier to track the metrics to completely remove after 2 minor releases or something, and it would be very simple to remove them

Reply:
I updated the PR so that the new metric names will be the recording rules and in 2 versions we would replace the metric names and remove the recording rules.
I did this since the help text is not visible for recording rule, only for metrics.